pkcs11/src/new/mod.rs

```diff
@@ -158,50 +166,36 @@ mod tests {
         // get mechanism
         let mechanism = Mechanism::RsaPkcsKeyPairGen;
 
-        let mut attr_true = pkcs11_sys::CK_TRUE;
-        let mut attr_true1 = pkcs11_sys::CK_TRUE;
-        let mut attr_false = pkcs11_sys::CK_FALSE;
-        let mut public_exponent: Vec<u8> = vec![0x01, 0x00, 0x01];
-        let mut modulus_bits: u64 = 1024;
+        let public_exponent: Vec<u8> = vec![0x01, 0x00, 0x01];
+        let modulus_bits: u64 = 1024;
 
         // pub key template
-        let mut pub_key_template = vec![
-            Attribute::Token(&mut attr_true),
-            Attribute::Private(&mut attr_false),
-            Attribute::PublicExponent(&mut public_exponent),
-            Attribute::ModulusBits(&mut modulus_bits),
+        let pub_key_template = vec![
+            Attribute::Token(Box::pin(CK_TRUE)),
```

Comment:
Is this OK for users of this crate to do? It's better than before but still a bit verbose to create those attributes. Ideally they should not have to care or know that `Pin` is used.

Maybe we could have constructors for each one of them like:
```rust
impl Attribute {
    pub fn new_token(val: bool) {
        Attribute::Token(Box::pin(if val { CK_TRUE } else { CK_FALSE }))
    }
}
```
but then should we still make the structure fully transparent (everything is public)?

Reply:
I propose we keep the structure transparent and do those things:
* abstract each `CK_` type into its own Rust abstraction under `types`. With conversions from `bool`, `u64` for native ones.
* use those abstracted types in the `Attribute` enum instead of the raw C types
* add constructors to easily build the different variant of `Attribute`.

Reply:
> This may sound somewhat dumb, but would it be an option to not hold a Pin here, but create it only when/where we need that address to be kept in place? E.g. when you call generate_key_pair with some list of attributes, we take each one and clone it from &Attribute -> Pin<Attribute> and then convert from there, smth like From<&Pin<Attribute>> for CK_ATTRIBUTE?

Yes it's possible, would make a lot of `clone` though

Reply:
For the operations that we've covered, is there a need to keep the attributes after you pass them to a function that consumes?

Reply:
I think you might want to keep ownership, specially when you sign or encrypt something.

Reply:
You mean to check something before you sign/encrypt? Neither signing nor encrypting require attributes

Reply:
Oh yes that is true I mixed up things in my head 🤯 
I was actually thinking of the data that is in vectors: like the key modulus for example, maybe in some cases you want to use it after calling an operation? There might be other attributes that we don't currently support that you also might want to keep after operations, like `CKA_VALUE`.

Reply:
Ok, well, I think that was my idea for "fixing" this messy "Box::pin" everywhere. 

Oh, another option is - if you make "abstracted" `BBOOL` and such types, you can obviously implement `From<BBool> for Pin<Box<BBool>>`. That's to say, I'm not a huge fan of having per-attribute-type constructors if they're public anyway...

Reply:
Let's that true, let's go with creating abstraction for every types and conversions to their pinned counterpart